weather_report/report_generator.py

```diff
@@ -0,0 +1,85 @@
+"""Responsible for displaying all weather reports
+"""
+from datetime import datetime
+
+from utils import value_error_decorator
+from weather_man import WeatherMan
+
+
+class ReportGenerator:
+    """Displays weather reports"""
+
+    def __init__(self, folder_name: str) -> None:
+        self._weather_man = WeatherMan(folder_name)
+
+    @value_error_decorator
+    def yearly_extremes_report(self, year_date: datetime) -> None:
+        """Prints year extreme values"""
+        print(f"\n=== Report for Year {year_date.year} === ")
+        (
+            highest_temp_record,
+            lowest_temp_record,
+            highest_humidity_record,
+        ) = self._weather_man.get_year_extremes(year_date)
+
+        print(
+            f'Highest: {highest_temp_record.max_temp}C on \
+            {highest_temp_record.date.strftime("%B %d")}'
+        )
+        print(
+            f'Lowest: {lowest_temp_record.min_temp}C on \
+            {lowest_temp_record.date.strftime("%B %d")}'
+        )
+        print(
+            f'Humidity: {highest_humidity_record.max_humidity}% on \
+            {highest_humidity_record.date.strftime("%B %d")}'
+        )
+
+    @value_error_decorator
+    def monthly_avg_report(self, year_month: datetime) -> None:
+        """Prints month average values"""
+        print(f"\n=== Report for {year_month.strftime('%B, %Y')} === ")
+        (
+            highest_temp_avg,
+            lowest_temp_avg,
+            mean_humidity_avg,
+        ) = self._weather_man.get_month_avg(year_month)
+
+        print(f"Highest Temperature Average: {highest_temp_avg}C")
+        print(f"Lowest Temperature Average: {lowest_temp_avg}C")
+        print(f"Mean Humidity Average: {mean_humidity_avg}%")
+
+    @value_error_decorator
+    def daily_month_report(self, year_month: datetime) -> None:
+        """Print each day max and min temps with bars"""
+        weather_data = self._weather_man.filter_data(year_month)
+        print(f"\n=== Report for {year_month.strftime('%B, %Y')} ===")
+        if not weather_data:
```

Comment:
a blank line before if condition will make the code more clean and readable. 
please add a blank line where required, i.e to separate similar logical lines, etc